### PR TITLE
Remove unused result in for statement

### PR DESCRIPTION
**Pull request by Arman-Hosseini**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Remove unused result in for statement
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18097)

### CODE REVIEW

File: controllers/admin/AdminCartRulesController.php [hunk 1]
```diff
@@ -171,9 +171,7 @@ public function postProcess()
                     $rule_group_id = 1;
                     if (is_array($rule_group_array)) {
                         // Empty for (with a ; at the end), that just find the first rule_group_id available in rule_group_array
-                        for ($rule_group_id = 1; in_array($rule_group_id, $rule_group_array); ++$rule_group_id) {
-                            42;
```
Comment (matks) [inline controllers/admin/AdminCartRulesController.php:175]: What is that 😅

**Comment by matks** [comment]
Hi @Arman-Hosseini thanks for the PR. I am sorry, honestly I dont understand neither the previous statement nor the new statement you wrote 😅it might be valid php syntax but it's unusual.

Can you please explain the change you introduced ?

Also, maybe if I cant understand this code, it means a lot more people will not be able to understand it either. So maybe we can replace this code by something simpler, easier (maybe less optimized), or add a comment explaining the logic ?

File: controllers/admin/AdminCartRulesController.php [hunk 2]
```diff
@@ -171,9 +171,7 @@ public function postProcess()
...
-                        }
```
Comment (Arman-Hosseini) [inline controllers/admin/AdminCartRulesController.php]: >     // Empty for (with a ; at the end), that just find the first rule_group_id available in rule_group_array 

@matks Now I look at the comment and think that the logic of the code is not correct. The comment states that it finds the first value available in the array.

**Review by PierreRambaud — APPROVED** [review]

File: controllers/admin/AdminCartRulesController.php [hunk 3]
```diff
@@ -171,9 +171,7 @@ public function postProcess()
...
+                        for (; in_array($rule_group_id, $rule_group_array); ++$rule_group_id);
```
Comment (PierreRambaud) [inline controllers/admin/AdminCartRulesController.php]: Should be done with a while for a better readability? :thinking:

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Wrong button, clicked too fast. Read comment :sweat_smile:

File: controllers/admin/AdminCartRulesController.php (hunk 2 above)
Comment (Arman-Hosseini) [reply controllers/admin/AdminCartRulesController.php]: @PierreRambaud What do you think here?

Comment (PierreRambaud) [reply controllers/admin/AdminCartRulesController.php]: I agree with you, the statement is useless, but if you're thinking algorithm, it should be a whil it's in array, then increase the var

File: controllers/admin/AdminCartRulesController.php (hunk 3 above)
Comment (Arman-Hosseini) [reply controllers/admin/AdminCartRulesController.php]: Looks good for transparency. But I have a better idea.
The logic of the code gives us The first incremental result that is not in the array.
Isn't this better? 
`$rule_group_id = end($rule_group_array) + 1;`

Comment (PululuK) [reply controllers/admin/AdminCartRulesController.php]: But ... why making it simple when we can make it complicated ? 😁

Comment (PierreRambaud) [reply controllers/admin/AdminCartRulesController.php]: I think they thought about something like this:
```php
$rule_group_array = [1, 2, 3, 4, 5, 9, 10];
for (; in_array($rule_group_id, $rule_group_array); ++$rule_group_id);
// => $rule_group_id = 6;
```

Comment (PierreRambaud) [reply controllers/admin/AdminCartRulesController.php]: Otherwise an autoincrement field should have been used

Comment (matks) [reply controllers/admin/AdminCartRulesController.php]: The very best code change would be something that, when we read it, we *immediately* understand *exactly* what is going on 😄. Here, with the current implementation, we are "I think it does this ..." ; "I think they wanted to do that ..." => such code is a bringer of complexity and bugs

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@Arman-Hosseini Thanks for the contribution. Could you fix PHPCS Fixer ? Thanks

File: controllers/admin/AdminCartRulesController.php (hunk 3 above)
Comment (PierreRambaud) [reply controllers/admin/AdminCartRulesController.php]: The whole legacy code is like that :sweat_smile:

**Comment by Arman-Hosseini** [comment]
@Progi1984 Logically there is no error.

**Comment by PululuK** [comment]
> @Progi1984 Logically there is no error.

Hi @Arman-Hosseini ... Thanks man !
To fix it you can just run` php ./vendor/bin/php-cs-fixer fix` from the root folder

**Comment by Arman-Hosseini** [comment]
> > @Progi1984 Logically there is no error.
> 
> Hi @Arman-Hosseini ... Thanks man !
> To fix it you can just run` php ./vendor/bin/php-cs-fixer fix` from the root folder

Hi @PululuK !
There is no difference here between $rule_group_id++ or ++$rule_group_id
But anyway it was done =))

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
No need QA. Thanks @Arman-Hosseini

**Pull request merged**